src/context/SearchQueryContext.tsx

```diff
@@ -25,23 +25,49 @@ export type SearchQueryContextType = {
   filterFields: Facet[]
 }
 
+// Create a context for sharing search-related data and methods across components
 export const SearchQueryContext = createContext<SearchQueryContextType | null>(null);
 
+/**
+ * React context provider designed to encapsulate and manage the state and logic associated with search functionality.
+ * It leverages the NextJS router to read and update the URL's query parameters for search queries
+ * and pagination, and uses React Query for fetching search results from an Elasticsearch backend.
+ *
+ * Flow:
+ * 1. Extracts search parameters (search query, filters, sorting, page, and results per page) from the URL's query parameters.
+ * 2. Utilizes useMemo to memoize calculations for search parameters and URLSearchParams for efficient query manipulation.
+ * 3. The `useSearch` hook fetches search results based on the current search parameters.
+ * 4. Provides two main functions, `makeQuery` and `handlePageChange`, to update the URL's query parameters
+ *    and thereby trigger new searches or page changes. These updates are performed through the NextJS router,
+ *    allowing for client-side navigation without full page reloads.
+ * 5. Any change to the URL's query parameters triggers a re-fetch of search results via `useSearch`,
+ *    updating the `queryResult` state with the new results.
+ * 6. The updated search state (current search query, search results, pagination information) is made available
+ *    to child components through the `SearchQueryContext`, enabling a reactive search UI that updates in response
+ *    to user actions and URL changes.
```

Comment:
These points are helpful, but I think they would be better placed in a general explanations file (markdown?) than a docstring which should give the reader and idea of the API/Interface of the component not some implementation details.

Reply:
In my opinion, context about the application must be in places that readers/developer will find it while trying to understand the implementation details of the codebase.
So I don't think that creating a markdown file somewhere else in order to explain what happens in the `SearchQueryProvider` is a good strategy.